### PR TITLE
Created flask app

### PR DESCRIPTION
**Pull request by Mujtaba-Akbar**
# Created flask app
- App routes to display data from server 
- Yearly extremes template
- Monthly average template
- Bar chart template
- Create user profile
- Show users
- Added tailwind for styling

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1 +1,2 @@
-__pycache__
\ No newline at end of file
+__pycache__
+node_modules
```
Comment (Itezaz-ul-Hassan) [inline .gitignore]: Always add new line at the end of file

File: actions.py [hunk 2]
```diff
@@ -5,8 +5,7 @@
 
 
 class YearlyExtremesAction(argparse.Action):
```
Comment (Itezaz-ul-Hassan) [inline actions.py:7]: Why do we need argparse.Action now? We're performing actions through flask app

File: app.py [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+
+
+@app.route('/show_users', methods=['GET'])
+def show_users():
+    users = load_users()
+    return render_template('show_users.html', users=users)
+
+
+if __name__ == '__main__':
+    app.run(debug=True)
```
Comment (Itezaz-ul-Hassan) [inline app.py]: We should always make a separate config file and define relevant settings there

File: report_generator.py [hunk 4]
```diff
@@ -90,21 +123,48 @@ def print_month_extremes_bar_chart(self):
...
                 int(max_temp) + RESET_COLOR + f" {int(max_temp)}C"
 
-    # generate bar for min temp in blue
+        # generate bar for min temp in blue
             min_bar = BLUE_COLOR + '+' * \
                 int(min_temp) + RESET_COLOR + f" {int(min_temp)}C"
-            print(f"{day:02} {max_bar}")
-            print(f"{day:02} {min_bar}")
+
+            # print(f"{day:02} {max_bar}")
```
Comment (Itezaz-ul-Hassan) [inline report_generator.py]: Remove extra comments

**Review by Itezaz-ul-Hassan — CHANGES_REQUESTED** [review]

File: actions.py (hunk 2 above)
Comment (Mujtaba-Akbar) [reply actions.py:7]: This is the code from the first task where command line was being used, so it's not being used for the django app. The first task's code is also in this repo, its refactored.

**Comment by Mujtaba-Akbar** [comment]
@Itezaz-ul-Hassan resolved, kindly check and merge.

**Review by Itezaz-ul-Hassan — APPROVED** [review]

**Pull request merged**